### PR TITLE
collection: format, fix cross-compile, cleanups, etc.

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gstreamer/bad/default.nix [hunk 1]
```diff
@@ -165,17 +168,21 @@ in stdenv.mkDerivation rec {
     libxml2
     libintl
     srt
+<<<<<<< HEAD
```
Comment (siraben) [inline pkgs/development/libraries/gstreamer/bad/default.nix:171]: Must be a `git` merge conflict artifact 😆

File: pkgs/development/libraries/vtk/generic.nix [hunk 2]
```diff
@@ -94,6 +94,6 @@ in stdenv.mkDerivation rec {
     maintainers = with maintainers; [ knedlsepp tfmoraes lheckemann ];
     platforms = with platforms; unix;
     # /nix/store/xxxxxxx-apple-framework-Security/Library/Frameworks/Security.framework/Headers/Authorization.h:192:7: error: variably modified 'bytes' at file scope
-    broken = if stdenv.isDarwin && (majorVersion == 7 || majorVersion == 8) then true else false;
+    broken = if stdenv.isDarwin && ((lib.versionsmajor majorVersion) == "7" || (lib.versions.major majorVersion) == "8") then true else false;
```
Comment (siraben) [inline pkgs/development/libraries/vtk/generic.nix:97]:
```suggestion
    broken = if stdenv.isDarwin && ((lib.versions.major majorVersion) == "7" || (lib.versions.major majorVersion) == "8") then true else false;
```

File: pkgs/development/tools/wrangler/default.nix [hunk 3]
```diff
@@ -13,15 +13,11 @@ rustPlatform.buildRustPackage rec {
...
+    ++ lib.optionals stdenv.isLinux [ pkg-config ];
 
   buildInputs = lib.optionals stdenv.isLinux [ openssl ]
-    ++ lib.optionals stdenv.isDarwin [
-      curl
-      darwin.apple_sdk.frameworks.Security
-      darwin.apple_sdk.frameworks.CoreServices
-      darwin.apple_sdk.frameworks.CoreFoundation
-    ];
+    ++ lib.optionals stdenv.isDarwin [ curl CoreFoundation CoreServices Security ];
```
Comment (siraben) [inline pkgs/development/tools/wrangler/default.nix:20]: Why is `curl` only needed on darwin?

**Review by siraben — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/wrangler/default.nix (hunk 3 above)
Comment (siraben) [reply pkgs/development/tools/wrangler/default.nix:20]: I can run a build to check on darwin.

File: pkgs/development/libraries/vtk/generic.nix [hunk 4]
```diff
@@ -94,6 +94,6 @@ in stdenv.mkDerivation rec {
     maintainers = with maintainers; [ knedlsepp tfmoraes lheckemann ];
     platforms = with platforms; unix;
     # /nix/store/xxxxxxx-apple-framework-Security/Library/Frameworks/Security.framework/Headers/Authorization.h:192:7: error: variably modified 'bytes' at file scope
-    broken = if stdenv.isDarwin && (majorVersion == 7 || majorVersion == 8) then true else false;
+    broken = if stdenv.isDarwin && (lib.versions.major majorVersion == "7" || lib.versions.major majorVersion == "8") then true else false;
```
Comment (siraben) [inline pkgs/development/libraries/vtk/generic.nix:97]:
```suggestion
    broken = stdenv.isDarwin && (lib.versions.major majorVersion == "7" || lib.versions.major majorVersion == "8");
```
Redundant if

File: pkgs/development/libraries/bootil/default.nix [hunk 5]
```diff
@@ -1,18 +1,8 @@
...
-    description = "Garry Newman's personal utility library";
-    homepage = "https://github.com/garrynewman/bootil";
-    # License unsure - see https://github.com/garrynewman/bootil/issues/21
-    license = lib.licenses.free;
-    maintainers = [ lib.maintainers.abigailbuccaneer ];
-    platforms = lib.platforms.all;
-    # Build uses `-msse` and `-mfpmath=sse`
-    badPlatforms = [ "aarch64-linux" ];
-  };
+  name = "bootil";
```
Comment (siraben) [inline pkgs/development/libraries/bootil/default.nix:4]:
```suggestion
  pname = "bootil";
```

File: pkgs/applications/misc/prusa-slicer/super-slicer.nix [hunk 6]
```diff
@@ -1,6 +1,4 @@
-{
-  lib, fetchFromGitHub, makeDesktopItem, prusa-slicer
-}:
+{ stdenv, lib, fetchFromGitHub, makeDesktopItem, prusa-slicer }:
```
Comment (siraben) [inline pkgs/applications/misc/prusa-slicer/super-slicer.nix:1]:
```suggestion
{ lib, fetchFromGitHub, makeDesktopItem, prusa-slicer }:
```
Redundant `stdenv`

Comment (siraben) [inline pkgs/applications/misc/prusa-slicer/super-slicer.nix:1]:
```suggestion
{ lib, fetchFromGitHub, makeDesktopItem, prusa-slicer }:
```
Redundant `stdenv`.

File: pkgs/development/libraries/vtk/generic.nix (hunk 4 above)
Comment (siraben) [inline pkgs/development/libraries/vtk/generic.nix:97]:
```suggestion
    broken = stdenv.isDarwin && (lib.versions.major majorVersion == "7" || lib.versions.major majorVersion == "8");
```
Redundant `if`.

**Review by siraben — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/wrangler/default.nix (hunk 3 above)
Comment (siraben) [reply pkgs/development/tools/wrangler/default.nix:20]: Without curl on darwin:
```
       >   = note: ld: library not found for -lcurl
       >           clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
       >           
       >
       > error: aborting due to previous error
       >
       > error: could not compile `wrangler`
       >
       > To learn more, run the command again with --verbose.
       For full logs, run 'nix log /nix/store/qja5q76kd8bjzj8y4f58d8g8wv5c7zww-wrangler-1.13.0.drv'.
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115197` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>poke</li>
<li>profanity</li>
</ul>
</details>
<details>

  <summary>3 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python37Packages.google-api-python-client: log was empty</li>
<li>python38Packages.google-api-python-client: log was empty</li>
<li>thefuck: log was empty</li>
  </ul>
</details>
<details>

  <summary>10 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.bpython: log was empty</li>
<li>python37Packages.curtsies: log was empty</li>
<li>python37Packages.pyte: <a href="https://termbin.com/lkdg">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/c0587ed06916e4dd78b3ec3f8513152c/raw/index.html">pretty log </a></li>
<li>python38Packages.bpython: log was empty</li>
<li>python38Packages.curtsies: log was empty</li>
<li>python38Packages.pyte: <a href="https://termbin.com/jdd3">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/0d581dbc97932d5ed3d470483f800031/raw/index.html">pretty log </a></li>
<li>python39Packages.bpython: log was empty</li>
<li>python39Packages.curtsies: log was empty</li>
<li>python39Packages.pyte: <a href="https://termbin.com/0nvm">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/97ca53e7b09fb7fa4277cf5db76fc549/raw/index.html">pretty log </a></li>
<li>termtosvg: log was empty</li>
  </ul>
</details>
<details>
<summary>
19 packages built:
</summary>
<ul>
<li>arachne-pnr</li>
<li>bootil</li>
<li>dejagnu</li>
<li>expect</li>
<li>gmad</li>
<li>gti</li>
<li>haskellPackages.nix-output-monitor</li>
<li>http-prompt</li>
<li>httpie</li>
<li>monotone</li>
<li>nix-output-monitor</li>
<li>pkgdiff</li>
<li>python37Packages.googleapis-common-protos</li>
<li>python38Packages.googleapis-common-protos</li>
<li>python39Packages.google-api-python-client</li>
<li>python39Packages.googleapis-common-protos</li>
<li>safe-rm</li>
<li>wdiff</li>
<li>wrangler</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
bootil:
<details>


warning: attribute-typo
A likely typo in `premakefile` argument was found, did you mean `makefile`?

Near pkgs/development/libraries/bootil/default.nix:31:3:

```
   |
31 |   premakefile = "projects/premake4.lua";
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
</details>
dejagnu:
<details>

warning: EvalError
Cannot evaluate attribute ‘dejagnu’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
expect:
<details>

warning: EvalError
Cannot evaluate attribute ‘expect’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
gmad:
<details>

warning: EvalError
Cannot evaluate attribute ‘gmad’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
gti:
<details>

Using CC=cc, CXX=c++, AR=ar or LD=cc prevents cross compiling.
Please use ``CC=${stdenv.cc.targetPrefix}cc``, ``CXX=${stdenv.cc.targetPrefix}c++``, ``AR=${stdenv.cc.targetPrefix}ar`` or ``LD=${stdenv.cc.targetPrefix}cc``.

</details>
haskellPackages.nix-output-monitor:
<details>

warning: EvalError
Cannot evaluate attribute ‘haskellPackages.nix-output-monitor’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
monotone:
<details>

warning: EvalError
Cannot evaluate attribute ‘monotone’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
nix-output-monitor:
<details>

warning: EvalError
Cannot evaluate attribute ‘nix-output-monitor’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
pkgdiff:
<details>

warning: EvalError
Cannot evaluate attribute ‘pkgdiff’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
python37Packages.bpython:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.google-api-python-client:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/google-api-python-client/default.nix:37:5:

```
   |
37 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python37Packages.googleapis-common-protos:
<details>


warning: unused-argument
Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/googleapis-common-protos/default.nix:6:3:

```
  |
6 | , pytestCheckHook
  |   ^
```
</details>
python37Packages.pyte:
<details>

build log of '/nix/store/l8g573qn2bzwspfwpzn6dqj61hf0na8j-python3.7-pyte-0.8.0' is not available
error: --- Error --- nix
build log of '/nix/store/l8g573qn2bzwspfwpzn6dqj61hf0na8j-python3.7-pyte-0.8.0' is not available
When evaluating attribute ‘python37Packages.pyte’:
warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyte/default.nix:21:5:

```
   |
21 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python38Packages.bpython:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.google-api-python-client:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/google-api-python-client/default.nix:37:5:

```
   |
37 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python38Packages.googleapis-common-protos:
<details>


warning: unused-argument
Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/googleapis-common-protos/default.nix:6:3:

```
  |
6 | , pytestCheckHook
  |   ^
```
</details>
python38Packages.pyte:
<details>

build log of '/nix/store/2xb9yi2qhx6jfpgfv76mcn61ck441f2a-python3.8-pyte-0.8.0' is not available
error: --- Error --- nix
build log of '/nix/store/2xb9yi2qhx6jfpgfv76mcn61ck441f2a-python3.8-pyte-0.8.0' is not available
When evaluating attribute ‘python38Packages.pyte’:
warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyte/default.nix:21:5:

```
   |
21 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python39Packages.bpython:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.google-api-python-client:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/google-api-python-client/default.nix:37:5:

```
   |
37 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python39Packages.googleapis-common-protos:
<details>


warning: unused-argument
Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/googleapis-common-protos/default.nix:6:3:

```
  |
6 | , pytestCheckHook
  |   ^
```
</details>
python39Packages.pyte:
<details>

build log of '/nix/store/wm4m8spdys9a56n4m9aawl3p4s9pzqk6-python3.9-pyte-0.8.0' is not available
error: --- Error --- nix
build log of '/nix/store/wm4m8spdys9a56n4m9aawl3p4s9pzqk6-python3.9-pyte-0.8.0' is not available
When evaluating attribute ‘python39Packages.pyte’:
warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/pyte/default.nix:21:5:

```
   |
21 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
safe-rm:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/system/safe-rm/default.nix:33:5:

```
   |
33 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
wdiff:
<details>

warning: EvalError
Cannot evaluate attribute ‘wdiff’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
wrangler:
<details>

warning: EvalError
Cannot evaluate attribute ‘wrangler’ in ‘/nix/store/im2yvpgaqr622y2vj2ljc2jwxavsrv1h-nixpkgs’.
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115197` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.virtualbox</li>
<li>linuxPackages_latest-libre.virtualbox</li>
</ul>
</details>
<details>
<summary>
57 packages built:
</summary>
<ul>
<li>arachne-pnr</li>
<li>bootil</li>
<li>dejagnu</li>
<li>expect</li>
<li>fmbt</li>
<li>gmad</li>
<li>gti</li>
<li>hashdeep</li>
<li>http-prompt</li>
<li>httpie</li>
<li>linuxPackages.virtualbox</li>
<li>linuxPackages_4_14.virtualbox</li>
<li>linuxPackages_4_19.virtualbox</li>
<li>linuxPackages_4_4.virtualbox</li>
<li>linuxPackages_4_9.virtualbox</li>
<li>linuxPackages_5_10.virtualbox</li>
<li>linuxPackages_5_11.virtualbox</li>
<li>linuxPackages_hardened.virtualbox</li>
<li>linuxPackages_latest_hardened.virtualbox</li>
<li>linuxPackages_latest_xen_dom0.virtualbox</li>
<li>linuxPackages_latest_xen_dom0_hardened.virtualbox</li>
<li>linuxPackages_lqx.virtualbox</li>
<li>linuxPackages_testing_bcachefs.virtualbox</li>
<li>linuxPackages_xen_dom0.virtualbox</li>
<li>linuxPackages_xen_dom0_hardened.virtualbox</li>
<li>linuxPackages_zen.virtualbox</li>
<li>monotone</li>
<li>nix-output-monitor</li>
<li>open-watcom-bin</li>
<li>pkgdiff</li>
<li>poke</li>
<li>profanity</li>
<li>python37Packages.bpython</li>
<li>python37Packages.curtsies</li>
<li>python37Packages.google-api-python-client</li>
<li>python37Packages.googleapis-common-protos</li>
<li>python37Packages.pyte</li>
<li>python38Packages.bpython</li>
<li>python38Packages.curtsies</li>
<li>python38Packages.google-api-python-client</li>
<li>python38Packages.googleapis-common-protos</li>
<li>python38Packages.pyte</li>
<li>python39Packages.bpython</li>
<li>python39Packages.curtsies</li>
<li>python39Packages.google-api-python-client</li>
<li>python39Packages.googleapis-common-protos</li>
<li>python39Packages.pyte</li>
<li>safe-rm</li>
<li>swtpm (swtpm-tpm2)</li>
<li>termtosvg</li>
<li>thefuck</li>
<li>viking</li>
<li>virtualbox</li>
<li>virtualboxHardened</li>
<li>virtualboxHeadless</li>
<li>virtualboxWithExtpack</li>
<li>wdiff</li>
</ul>
</details>

**Comment by sternenseemann** [comment]
This PR introduced breaking changes to the public interface of nixpkgs mixed in with trivial formatting changes and cleanups and was merged within little over 24h after all but one review (which was done before the breakage was introduced).

I can sympathize with the desire to bring the attribute names in `pythonPackages` in line with the upstream names and PyPi, but downstream users depend on these attributes and removing them introduces unnecessary and unmanaged breakage (for example [this is how I got wind of this PR](https://code.tvl.fyi/commit/third_party/lieer/default.nix?id=90fdd7f23d6cc0d1374332d49218aef9a0d6a52f)). Maybe the renames should have waited until we have [alias support for pythonPackages](https://github.com/NixOS/nixpkgs/pull/116367)?

**Pull request merged**